pkg/util/cert_pool.go

```diff
@@ -17,28 +17,45 @@ limitations under the License.
 package util
 
 import (
+	"crypto/sha256"
 	"crypto/x509"
 	"encoding/pem"
 	"fmt"
 	"time"
 )
 
 // CertPool is a set of certificates.
-type certPool struct {
-	certificates  []*x509.Certificate
-	filterExpired bool
+type CertPool struct {
+	certificatesHashes map[[32]byte]struct{}
+	certificates       []*x509.Certificate
+	filterExpired      bool
+}
+
+type Options func(*CertPool)
+
+func WithFilteredExpiredCerts(filterExpired bool) Options {
+	return func(cp *CertPool) {
+		cp.filterExpired = filterExpired
+	}
 }
 
 // newCertPool returns a new, empty CertPool.
-func newCertPool(filterExpired bool) *certPool {
-	return &certPool{
-		certificates:  make([]*x509.Certificate, 0),
-		filterExpired: filterExpired,
+func NewCertPool(options ...Options) *CertPool {
+	var certPool = &CertPool{
+		certificates:       make([]*x509.Certificate, 0),
+		filterExpired:      true,
```

Comment:
```suggestion
```

Or is this required for some reason?

Reply:
No, there is no any reason to set `filterExpired` to `true`.
Fixed.